autoload/go/doc.vim

```diff
@@ -77,14 +77,45 @@ endfunction
 
 function! s:GodocView(newposition, position, content) abort
   " popup window
-  if go#config#DocPopupWindow() && exists('*popup_atcursor') && exists('*popup_clear')
-    call popup_clear()
-
-    call popup_atcursor(split(a:content, '\n'), {
-          \ 'padding': [1, 1, 1, 1],
-          \ 'borderchars': ['-','|','-','|','+','+','+','+'],
-          \ "border": [1, 1, 1, 1],
-          \ })
+  if go#config#DocPopupWindow()
+    if exists('*popup_atcursor') && exists('*popup_clear')
+      call popup_clear()
+
+      call popup_atcursor(split(a:content, '\n'), {
+            \ 'padding': [1, 1, 1, 1],
+            \ 'borderchars': ['-','|','-','|','+','+','+','+'],
+            \ "border": [1, 1, 1, 1],
+            \ })
+    elseif has('nvim') && exists('*nvim_open_win')
+      let lines = split(a:content, '\n')
+      let height = 0
+      let width = 0
+      for line in lines
+        let lw = strdisplaywidth(line)
+        if lw > width
+          let width = lw
+        endif
+        let height += 1
+      endfor
+      let width += 1 " right margin
```

Comment:
Some kind of maximum is needed for height so that large godoc doesn't take up the whole screen.